File: Fintranet Library/Web/FinLib.Web/wwwroot/Pages/core/models/dtos/UserDto.ts





interface IUserDto extends IUpdatableDto 
{   
    firstName: string;
    lastName: string;
    fullName: string;
    gender: Gender;
    birthDate: Date;
    mobile: string;
    imageUrl: string;
    userName: string;
    password: string;
    email: string;
    isActive: boolean;
    lastLoggedInTime: Date;
    lockoutDescription: string;
    userRoles: IUserRoleDto[];
}

File: Fintranet Library/Core/FinLib.Models/Dto/SEC/UserDto.cs
using FinLib.Models.Enums;

namespace FinLib.Models.Dtos.SEC
{
    public class UserDto : Base.Dto.UpdatableDto
    {
        #region Personal Related
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName { get { return FirstName + " " + LastName; } }

        public Gender? Gender { get; set; }
        public DateTime? BirthDate { get; set; }
        
        public string Mobile { get; set; }
        public string ImageUrl { get; set; }
        #endregion


        #region Identity Related

        public string UserName { get; set; }
        public string Password { get; set; }
        public string Email { get; set; }
        public bool IsActive { get; set; }
        public DateTime? LastLoggedInTime { get; set; }
        public string LockoutDescription { get; set; }

        public List<UserRoleDto> UserRoles { get; set; }

        #endregion

        public override string ToString()
        {
            return UserName;
        }
    }
}